src/main/java/services/car/CarComfortService.java

```diff
@@ -1,44 +1,98 @@
-import repository.TransactionManager;
 import services.CrudGenericInterface;
 
 import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.UUID;
 
 @Slf4j
-@RequiredArgsConstructor
-public class CarComfortService implements CrudGenericInterface<CarComfort>, TransactionManager {
+@AllArgsConstructor
+public class CarComfortService implements CrudGenericInterface<CarComfort>, Transactionable {
     private CarComfortRepository carComfortRepository;
 
     @Override
-    public void insert(CarComfort carComfort) {
-        /*wrapIntoTransaction(() -> {
-            carComfortRepository.insert(null);
-            carComfortRepository.getAll();
-            carComfortRepository.delete(null);
-        });*/
+    public CarComfort insert(CarComfort carComfort) {
+        log.info("Trying to insert new car comfort");
+        validateCarComfort(carComfort);
+        CarComfort c = wrapIntoTransaction(() -> carComfortRepository.insert(carComfort));
+        log.info("Car comfort has been inserted successfully");
+        return c;
     }
 
     @Override
     public List<CarComfort> getAll() {
-        return null;
+        log.info("Trying to get all car comforts");
+        return carComfortRepository.getAll();
     }
 
     @Override
     public CarComfort getById(UUID id) {
-        return null;
+        log.info("Trying to get car comfort by id");
+        if (Objects.isNull(id)) {
+            log.error("Can not find car comfort. Car comfort id must be not null");
+            throw new CarRentalException("Car comfort id must be NOT null");
+        }
+        try {
+            return carComfortRepository.getById(id);
+        } catch (CarRentalException exception) {
+            log.warn("Car comfort with id {} not found", id);
+            throw new CarRentalException("Car comfort by SUCH an id must be NOT null");
+        }
     }
 
     @Override
-    public void update(CarComfort carComfort) {
-        return;
+    public CarComfort update(CarComfort carComfort) {
+        log.info("Trying to update car comfort {}", carComfort);
+        if(!isCarComfortByIdExist(carComfort)){
+            log.warn("Car comfort with id {} not found", carComfort.getId());
+            throw new CarRentalException("Car comfort by such an id must be NOT null");
+        }
+        validateCarComfort(carComfort);
+        CarComfort c = wrapIntoTransaction(() -> carComfortRepository.update(carComfort));
+        log.info("Car comfort has been updated successfully");
+        return c;
     }
 
     @Override
-    public void delete(UUID id) {}
+    public void delete(UUID id) {
+        if (Objects.isNull(id)) {
+            log.error("Can not find car comfort. Car comfort id must be not null");
+            throw new CarRentalException("Can't remove the car comfort, its id must be NOT null");
+        }
+        try {
+            CarComfort carComfort = carComfortRepository.getById(id);
```

Comment:
Could you please remove try-catch and getById

Reply:
We can change the flow like this:
get car comfort by id
set inactive date
update car comfort

With this approach we do not need one more method for delete, and also we can return boolean here